src/t_cose_crypto.h

```diff
@@ -154,6 +156,137 @@ t_cose_crypto_sig_size(int32_t            cose_algorithm_id,
+ *         Successfully produced the HPKE algorithm structure.
+ * \retval T_COSE_ERR_UNSUPPORTED_KEY_EXCHANGE_ALG
+ *         The supported key exchange algorithm is not supported.
+ */
+
+enum t_cose_err_t
+t_cose_crypto_convert_hpke_algorithms(
+                int32_t                           hpke_cose_algorithm_id,
+                struct t_cose_crypto_hpke_suite_t *hpke_suite,
+                size_t                            *key_bitlen,
+                int64_t                           *cose_algorithm_id);
+
+/**
+ * \brief Exports the public key
+ *
+ * \param[in] key               Handle to key
+ * \param[in] pk_buffer         Pointer and length of buffer into which
+ *                              the resulting public key is put.
+ * \param[out] pk               Public Key
+ *
+ * \retval T_COSE_SUCCESS
+ *         Successfully exported the public key.
+ * \retval T_COSE_ERR_PUBLIC_KEY_EXPORT_FAILED
+ *         The public key export operation failed.
+ */
+enum t_cose_err_t
+t_cose_crypto_export_public_key(struct t_cose_key      key,
+                                struct q_useful_buf    pk_buffer,
+                                size_t                *pk_len);
+
+/**
+ * \brief HPKE Encrypt Wrapper
+ *
+ * \param[in] suite               HPKE ciphersuite
+ * \param[in] pkR                 pkR buffer
+ * \param[in] pkE                 pkE buffer
+ * \param[in] plaintext           Plaintext buffer
+ * \param[in] ciphertext          Ciphertext buffer
+ * \param[out] ciphertext_len     Length of the produced ciphertext
+ *
+ * \retval T_COSE_SUCCESS
+ *         HPKE encrypt operation was successful.
+ * \retval T_COSE_ERR_HPKE_ENCRYPT_FAIL
+ *         Encrypt operation failed.
+ */
+enum t_cose_err_t
+t_cose_crypto_hpke_encrypt(struct t_cose_crypto_hpke_suite_t  suite,
+                           struct q_useful_buf_c              pkR,
+                           struct t_cose_key                  pkE,
+                           struct q_useful_buf_c              plaintext,
+                           struct q_useful_buf                ciphertext,
+                           size_t                             *ciphertext_len);
+
+
+/**
+ * \brief HPKE Decrypt Wrapper
+ *
+ * \param[in] cose_algorithm_id   COSE algorithm id
+ * \param[in] pkE                 pkE buffer
+ * \param[in] pkR                 pkR key
+ * \param[in] ciphertext          Ciphertext buffer
+ * \param[in] plaintext           Plaintext buffer
+ * \param[out] plaintext_len      Length of the returned plaintext
+ *
+ * \retval T_COSE_SUCCESS
+ *         HPKE decrypt operation was successful.
+ * \retval T_COSE_ERR_UNSUPPORTED_KEY_EXCHANGE_ALG
+ *         An unsupported algorithm was supplied to the function call.
+ * \retval T_COSE_ERR_HPKE_DECRYPT_FAIL
+ *         Decrypt operation failed.
+ */
+enum t_cose_err_t
+t_cose_crypto_hpke_decrypt(int32_t                            cose_algorithm_id,
+                           struct q_useful_buf_c              pkE,
```

Comment:
pKE should be a struct t_cose_key. 

To make that work there should be a function to decode a COSE_Key into a struct t_cose_key that replaces a lot of code
in t_cose_encrypt_dec() and is part of the general support for encoding and decoding COSE_Key.